torch/csrc/jit/export.cpp

```diff
@@ -241,180 +197,546 @@ void encodeBlock(onnx::GraphProto * p_g, Block *b,
+    auto node_proto = graph_proto->add_node();
+    EncodeMethod(node_proto, method.value, prefix);
+  }
+
+  for (auto &submodule : module->get_modules()) {
+    EncodeMethods(graph_proto, submodule.value.module, prefix + submodule.key + ".");
+  }
+}
+
+void ModuleEncoder::EncodeMethod(
+    onnx::NodeProto *node_proto,
+    const std::unique_ptr<script::Method> &method,
+    const std::string prefix) {
+  node_proto->set_name(prefix + method->name());
+
+  // Store member_inputs of Method in input
+  for (auto &member_input : method->params()) {
+    auto it = parameter_map_.find(member_input);
+    JIT_ASSERT(it != parameter_map_.end());
+    node_proto->add_input(it->second);
+  }
+
+  auto attr_proto = node_proto->add_attribute();
+  attr_proto->set_type(onnx::AttributeProto_AttributeType_GRAPH);
+
+  for (auto node : method->graph()->nodes()) {
     IR_IF(node, PythonOp)
       auto py_node = static_cast<torch::jit::PythonOp*>(value);
-      FAIL_EXPORT(
+      throw std::runtime_error(
           "Couldn't export Python operator " + py_node->name() +
-          "\n\nDefined at:\n" + getNodeStackTraceString(node))
-    IR_ELSE()
-      // Special error messages for certain types of operators
-      if (node->kind() == aten::expand) {
-        FAIL_EXPORT(
-            "Could not export a broadcasted operation; ONNX likely does not support this form of broadcasting.\n\nBroadcast occurred at:\n" +
-            getNodeStackTraceString(node));
-      }
-      if (node->kind() == prim::PackPadded || node->kind() == prim::PadPacked) {
-        FAIL_EXPORT(
-            "Cannot export individual pack_padded_sequence or pad_packed_sequence; these operations must occur in pairs.\n\nUsage of this operation occurred at:\n" +
-            getNodeStackTraceString(node));
-      }
-      bool is_aten_fallback = operator_export_type == onnx::OperatorExportTypes::ONNX_ATEN_FALLBACK;
-      if (!node->kind().is_onnx() && !is_aten_fallback && node->kind() != prim::Undefined) {
-        FAIL_EXPORT(
-            "Couldn't export operator " + node->kind().toDisplayString() + "\n\nDefined at:\n" +
-            getNodeStackTraceString(node));
-      }
+          "\n\nDefined at:\n" + getNodeStackTraceString(node));
     IR_END()
-#undef FAIL_EXPORT
   }
+  EncodeBlock(attr_proto->mutable_g(), method->graph()->block(), {});
 }
 
+void ModuleEncoder::EncodeTensor(
+    onnx::TensorProto *tensor_proto,
+    const at::Tensor &tensor,
+    const at::optional<std::string> external_ref) {
+  for(auto d : tensor.sizes()) {
+    tensor_proto->add_dims(d);
+  }
+  tensor_proto->set_data_type(ATenTypeToOnnxType(tensor.type().scalarType()));
+  auto t = tensor.contiguous().toBackend(at::kCPU);
+
+  // The int64_data field stores offset, and strides
+  tensor_proto->add_int64_data(tensor.storage_offset());
+  for (auto &d : tensor.strides()) {
+    tensor_proto->add_int64_data(d);
+  }
+
+  auto storage_ptr = tensor.data_ptr();
```

Comment:
hm, I thought data_ptr returns you the shared buffers, but it won't catch the views of tensors (if one of them points to a subset of another). Do we want to support that?